src/dsm/headers/Agent.hpp

```diff
@@ -25,26 +25,29 @@ namespace dsm {
   private:
     Itinerary<Id> m_itinerary;
     double m_speed;
+    double m_position;
```

Comment:
Suppose we want to create two models, one of the first and one of the second order.
In one vase this parameter should represent a time penalty, i.e. a `uint` while in the second case it should be a `double` representing the position on the street (or the length minus the position if we care about the "missing" part of the journey).
I would template the parameter changing its name